src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -131,7 +131,7 @@ protected virtual void OpenPinCore(int pinNumber)
     /// </summary>
     /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
     /// <param name="mode">The mode to be set.</param>
-    public GpioPin OpenPin(int pinNumber, PinMode mode)
+    public virtual GpioPin OpenPin(int pinNumber, PinMode mode)
```

Comment:
[Triage] Let's try to reduce number of virtual OpenPin methods, if possible only OpenPinCore

Reply:
I've been looking at overwriting only the OpenPinCore but the issue is that the core logic with adding the pin into the lopen list and adding the gpio pin as well is done in the OpenPin. And that is what I want to override. And as the function returns a pin that won't be store properly otherwise, I don't really have a nice option for that!

Reply:
Fine for me. I mean if there's an use if this is virtual, I don't bother changing it.

Reply:
I think I found a good compromise and reviewed a bit the internal elements. So, I only made one of the OpenPin virtual, removed the others, reused some of the logic from the GpioPin dictionary. So it's cleaner.